NextcloudTalk/UserStatusMessageSwiftUIView.swift

```diff
@@ -0,0 +1,369 @@
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+import NextcloudKit
+
+struct UserStatusMessageSwiftUIView: View {
+
+    @Environment(\.dismiss) var dismiss
+    @Binding var changed: Bool
+    @State var showClearAtAlert: Bool = false
+
+    @State private var changedStatusFromPredif: Bool = false
+    @State private var customStatusSelected: Bool = false
+    @State private var selectedPredifinedStatus: NKUserStatus?
+    @State private var statusPredefinedStatuses: [NKUserStatus] = []
+
+    @State private var selectedIcon: String = ""
+    @State private var selectedMessage: String = ""
+    @State private var selectedClearAt: Double = 0
+    @State private var selectedClearAtString: String = ""
+
+    @State private var isLoading: Bool = true
+    @FocusState private var textFieldIsFocused: Bool
+    @State private var showErrorAlert = false
+    @State private var errorAlertTitle = ""
+    @State private var errorAlertMessage = ""
+
+    let clearAtOptions = [
+        NSLocalizedString("Don't clear", comment: ""),
+        NSLocalizedString("30 minutes", comment: ""),
+        NSLocalizedString("1 hour", comment: ""),
+        NSLocalizedString("4 hours", comment: ""),
+        NSLocalizedString("Today", comment: ""),
+        NSLocalizedString("This week", comment: "")
+    ]
+
+    init(changed: Binding<Bool>) {
+        _changed = changed
+    }
+
+    var body: some View {
+        ZStack(alignment: .bottom) {
+            if isLoading {
+                ProgressView()
+                    .tint(.secondary)
+            } else {
+                List {
+                    Section(header: Text(NSLocalizedString("What is your status?", comment: ""))) {
+                        HStack(spacing: 10) {
+                            EmojiTextFieldWrapper(placeholder: "😀", text: $selectedIcon)
+                                .frame(maxWidth: 23)
+                                .opacity(selectedIcon.isEmpty ? 0.5 : 1.0)
+                                .onChange(of: selectedIcon) { newString in
+                                    if changedStatusFromPredif {
+                                        customStatusSelected = false
+                                    } else {
+                                        customStatusSelected = true
+                                    }
```

Comment:
```suggestion
                                    customStatusSelected = !changedStatusFromPredif
```

But might not be neccessary at all (the comments above)